core/src/main/scala/org/apache/spark/Dependency.scala

```diff
@@ -76,6 +78,18 @@ class ShuffleDependency[K, V, C](
 
   override def rdd: RDD[Product2[K, V]] = _rdd.asInstanceOf[RDD[Product2[K, V]]]
 
+  /**
+   * The key, value and combiner classes are serialized so that shuffle manager
+   * implementation can use the information to build
+   */
+  val keyClassName: String = reflect.classTag[K].runtimeClass.getName
```

Comment:
Here's [an example of how I use this in the Parquet shuffle manager] (https://github.com/apache/spark/pull/7265/files#diff-2bbcf6433bd1ea2072d02cfb4a376985R53) to create a schema for the (key, value) or (key, combiner) pairs for the shuffle files.